### PR TITLE
Fix reboot time on salt-ssh client (bsc#1197591)

### PR DESCRIPTION
**Pull request by mbussolotto**
## What does this PR change?
ssh push reboot action has a valid response which contains the `shutdown` command, but the action needs to remain pending in order to execute `mgractionchains.resume` correctly. The problem does not happen with salt minion because the job `mgractionchains.resume` is executing after startup.

## GUI diff

No difference.

- [x] **DONE**

## Documentation

- [x] **DONE**

## Test coverage
- No tests: **add explanation**
- No tests: already covered
- Unit tests were added
- Cucumber tests were added

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/17388

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests" 
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by lucidd — APPROVED** [review]

File: java/code/src/com/suse/manager/webui/services/SaltServerActionService.java [hunk 1]
```diff
@@ -2523,19 +2523,20 @@ public void executeSSHAction(Action action, MinionServer minion, boolean forcePk
...
-                        // reboot needs special handling in case of ssh push
+                        /* bsc#1197591 ssh push reboot has an response that is not a failure but the action needs to
+                        *  remain pending, in this way SSHPushDriver::getCandidates can schedule a reboot correctly
+                        */
                         if (action.getActionType().equals(ActionFactory.TYPE_REBOOT) &&
-                            sa.getStatus().equals(ActionFactory.STATUS_QUEUED)) {
-                        // if the status is already PICKED_UP, don't change it
-                        // if the status is FAILED or COMPLETED, don't change it
+                                sa.getStatus().equals(ActionFactory.STATUS_QUEUED) ||
+                                sa.getStatus().equals(ActionFactory.STATUS_PICKED_UP)) {
```
Comment (mcalmer) [inline java/code/src/com/suse/manager/webui/services/SaltServerActionService.java:2531]: hmm, I think we had a real problem when the status is already "PICKED_UP" to change the pickup time and the updateServerInfo again. I am not sure if this break things again

**Review by mcalmer — COMMENTED** [review]
Let's discuss

**Comment by mbussolotto** [comment]
running  uyuni-prs-ci-tests

**Review by mcalmer — APPROVED** [review]
Looks better now. If the testsuite does not show any problems because of this I think we can go with it

**Comment by mbussolotto** [comment]
CI run with new test https://ci.suse.de/view/Manager/view/Uyuni-PRs/job/uyuni-prs-ci-tests/1250/

**Review by srbarrios — APPROVED** [review]

**Comment by srbarrios** [comment]
LGTM, test code side.

**Review by srbarrios — APPROVED** [review]

**Comment by mbussolotto** [comment]
CI test https://ci.suse.de/view/Manager/view/Uyuni-PRs/job/uyuni-prs-ci-tests/1269/

**Comment by srbarrios** [comment]
> CI test https://ci.suse.de/view/Manager/view/Uyuni-PRs/job/uyuni-prs-ci-tests/1269/

WOW All green :)

**Pull request merged**